### PR TITLE
Dpx

### PR DESCRIPTION
**Pull request by ThomasdenH**
A refactor of a part of dpx. This removes unsafe from a couple of functions.

### CODE REVIEW

File: dpx/src/specials/util.rs [hunk 1]
```diff
@@ -764,11 +766,11 @@ pub(crate) unsafe fn spc_util_read_blahblah(
                     }
                 }
                 "page" => {
-                    let mut page: f64 = 0.;
-                    if !page_no.is_null() && spc_util_read_numbers(&mut page, 1i32, ap) == 1i32 {
-                        *page_no = page as i32
+                    let page: ArrayVec<[_; 1]> = read_numbers(ap);
+                    if !page_no.is_null() && page.len() == 1 {
```
Comment (burrbull) [inline dpx/src/specials/util.rs]: Wrong sequence of calculations.
`read_numbers` has not be calculated if `page_no.is_null()`

File: dpx/src/specials/util.rs [hunk 2]
```diff
@@ -805,26 +807,24 @@ pub(crate) unsafe fn spc_util_read_blahblah(
...
-            && (has_scale != 0 || has_xscale != 0 || has_yscale != 0 || has_rotate != 0)
-        {
+        } else if has_matrix && (has_scale || has_xscale || has_yscale || has_rotate) {
             spc_warn!(spe, "Can\'t supply transform matrix along with scales or rotate. Ignore scales and rotate.");
         }
     }
-    if has_matrix == 0 {
-        make_transmatrix(&mut p.matrix, 0.0f64, 0.0f64, xscale, yscale, rotate);
+    if has_matrix {
+        make_transmatrix(&mut p.matrix, 0.0, 0.0, xscale, yscale, rotate);
```
Comment (burrbull) [inline dpx/src/specials/util.rs]: `!has_matrix`

**Comment by burrbull** [comment]
And rebase instead of merge, please.

**Comment by ThomasdenH** [comment]
I addressed the comments and rebased. Any files the project is currently focused on or is anything welcome?

**Pull request merged**